### PR TITLE
Us902

### PR DESCRIPTION
**Pull request by dzadroga**
US 902 Completes the Property Management UI

Tasks Completed:
- Build Property Management UI screen and prefab
- Implement property list display (upgrades state, mortgage state, cost information)
- Implement upgrade/downgrade button logic with dynamic interactivity
- Implement mortgage/unmortgage button logic with dynamic interactivity
- Wire actions to player data and property state change events

### CODE REVIEW

File: Assets/Scripts/UI/PropertyManagement/PropertyManagementRowUI.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    }
+
+    private void OnDowngradeClicked()
+    {
+         if (property is not PropertySpaceData streetProperty)
+            return;
+
+        if (streetProperty.GetCurrentUpgradeLevel() > 0)
+        {
+            streetProperty.SetUpgradeLevel(streetProperty.GetCurrentUpgradeLevel() - 1);
```
Comment (hdathert) [inline Assets/Scripts/UI/PropertyManagement/PropertyManagementRowUI.cs]: we need to make sure that when we sell property, we're actually adding the money back to the player... right now its just destroying upgrades without any benefit to the player.

**Review by hdathert — CHANGES_REQUESTED** [review]
I have a few questions:

First, I'm not seeing the mechanism for showing the UI- I see a `Show(Player)` method, but I'm not sure how thats meant to be hooked in.

I established a UI pattern in `PurchasePropertyUI` that relies on `UIActivationEvent` to open- the `HumanPlayerController` already is hooked into it and it can be extended with special context payloads. You can respond to the upgrade/downgrade/mortgage/unmortgage etc with `UIActionEvent`, that `HumanPlayerController` is also listening to. That way communication is via event, and the HPC can send new activation events to refresh the UI based on the result of actions from the property manager.

Going that route allows us to package all of the UI events into single channels, rather than continuing to balloon the number of channels that we have to feed into our player controllers.

That being said, it looks like things like mortgage are being handled directly on player methods, which is certainly a way to do it- we don't have some static calculator that is checking if thats valid or not, which I guess is fine, considering where we're at in the project.

So I guess we need to decide where these decisions are happening and how we're moving information between HPC and the UI. My assumption on how this would work is we'd have HPC -activate-> PropertyManagementUI, buttons would be actions -> HPC, which would verify and resolve and then update the PMUI with another activate.

Is that feasible, or did you have something else in mind?

If you have questions about anything I just said, definitely reach out!

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #902 and Tasks #930 to #934  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

**Comment by kescob0825** [comment]
# Code Review Checklist

**Completed for**: US #902
**Code Review Completed By**: Karl Escobar

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---
